cpp/include/cugraph/detail/graph_functions.cuh

```diff
@@ -183,7 +182,7 @@ gather_local_edges(
   GraphViewType const& graph_view,
   const rmm::device_uvector<typename GraphViewType::vertex_type>& active_majors_in_row,
   const rmm::device_uvector<gpu_t>& active_major_gpu_ids,
-  EdgeIndexIterator edge_index_first,
+  rmm::device_uvector<typename GraphViewType::edge_type>& minor_indices,
```

Comment:
The change is motivated by the requirement that the indices be returned too. Because there is a filtering happening in the sources, destinations, the same filtering must apply to the indices. This is a prim not exposable to API. The modification in place avoids an additional copy. This was authored by @kaatish , and we agreed to this format.

Reply:
OK.  I notice in the call to this function that the parameter passed as minor_indices is never used again.  Makes me wonder why it needs to be modified in place and maintained.

Reply:
It's part of the output that GNN wants.

Reply:
@ChuckHastings gather_local_edges does not return minor_indices again.
This is what it returns :
`  return std::make_tuple(std::move(majors), std::move(minors), std::move(minor_gpu_ids));`
minor_indices is passed as a reference because it is in/out param.

Reply:
Yes.  So can we clarify what is the difference between the `minor_indices` that is the input/output parameter and the `minors` that is returned as part of the tuple.

Reply:
Just a suggestion, mixing in/out parameters & return values might be a bit confusing, and can't we take `minor_indices` as an R-value return minor_indices as well? (instead of updating in-place).

Reply:
> can't we take minor_indices as an R-value return minor_indices as well

That would require an avoidable copy / move. The indices already exist, they just need to be filtered. You can pass them as `const&` but then in order to filter them you need a copy; i.e., extra storage + a move (at least; via RVO). We can avoid that, especially because this can be a large vector.

Reply:
I am not sure I am getting what you're saying.

If we're taking `minor_indices` as an **R-value**, we can modify the variable and return the variable.
```
...
rmm::device_uvector<typename GraphViewType::edge_type>&& minor_indices,
...
{
  filter(minor_indices);
  ...
  return std::make_tuple(..., std::move(minor_indices), ...);
}
```

Am I missing something?

Reply:
@seunghwak , unless you meant passing `minor_indices` as rvalue and then returning it as part of the tuple. IOW, did you mean:
```
vector foo(vector&& minor_indices){
  // blah... modify minor_indices;
  return minor_indices;
}
```
That only involves moves. That's what you meant?

Reply:
Yes!!! This wasn't possible when we were passing a pointer but now we're passing a device_uvector, and this is now possible.

Reply:
Okay. Let me do that.

Reply:
Done.

Reply:
Also addressed the naming issue and corresponding signature and dox.